tools/src/main/python/opengrok_tools/utils/mirror.py

```diff
@@ -393,18 +401,43 @@ def get_mirror_retcode(ignore_errors, value):
     return value
 
 
-def mirror_project(config, project_name, check_changes, uri,
-                   source_root, headers=None, timeout=None):
+def process_outgoing(repos, project_name):
+    """
+    Detect and strip any outgoing changes for the repositories.
+    :param repos: list of repository objects
+    :param project_name: name of the project
+    :return: if any of the repositories had to be reset
+    """
+
+    logger = logging.getLogger(__name__)
+    logger.info("Checking outgoing changes for project {}".format(project_name))
+
+    ret = False
+    for repo in repos:
+        logger.debug("Checking outgoing changes for repository {}".format(repo))
+        if repo.strip_outgoing():
+            logger.debug('Repository {} in project {} had outgoing changes stripped'.
+                         format(repo, project_name))
+            ret = True
+
+    return ret
+
+
+def mirror_project(config, project_name, check_changes, check_outgoing, uri,
+                   source_root, headers=None, timeout=None, api_timeout=None):
     """
     Mirror the repositories of single project.
     :param config global configuration dictionary
     :param project_name: name of the project
     :param check_changes: check for changes in the project or its repositories
      and terminate if no change is found
+    :param check_outgoing: check for outgoing changes in the repositories of the project,
```

Comment:
Why the change to `check_outgoing`? To me `check` has neutral meaning (it's just a check - whatever :D ) whereas `strip` is something I'd be more aware of.

Reply:
yep, check is not fitting; renamed.